utils/hare_cov/hare_coverage.py

```diff
@@ -0,0 +1,355 @@
+
+    def comparator(self, dir_name: str) -> bool:
+        """
+        Compares directory name with a set pattern.
+        :param dir_name: directory name
+        :return: boolean
+        """
+        return dir_name == self.pattern
+
+    def omit_dir(self, dir_name: str) -> bool:
+        """
+        Filter out the omitted directories.
+        :param dir_name: directory name
+        :return: boolean
+        """
+        for word in self.omit_kwargs:
+            if dir_name.find(word) != -1:
+                return True
+        return False
+
+    def search_dir(self, dir_path: str) -> List[str]:
+        """
+        Strategy for scanning directories.
+        :param dir_path: directory path
+        :return: List of the file match to given pattern
+        """
+        result_list: List[str] = []
+        for sub_dir, dirs, _ in walk(dir_path):
+            for directory in dirs:
+                directory_path: str = sub_dir + sep + directory
+                if self.omit_dir(directory_path):
+                    continue
+                result_list.extend([ele for ele in
+                                    self.search_dir(directory_path)
+                                    if ele not in result_list])
+                if self.callback(directory):
+                    if directory_path not in result_list:
+                        result_list.append(directory_path)
+        return result_list
+
+
+class Utility:
+    """Here is the utility class which has static utility helper methods."""
+
+    def run_cmd(self, command: str, delimiter: Optional[str] = None,
+                max_split: int = -1) -> None:
+        """
+        Execute command and return the result.
+        :param command: Command in string format
+        :param delimiter: delimiter (default is space)
+        :param max_split: maximum number split
+        :return: tuple(stdout, stderr)
+        """
+        logging.debug(f"Executing Command : {command!a}")
+        process: Popen[bytes] = Popen(
+            command.split(sep=delimiter, maxsplit=max_split),
+            stdout=PIPE,
+            stderr=PIPE)
+
+        out, err = process.communicate()
+        if process.returncode:
+            raise Exception(
+                f'Command {command!a} exited with error code '
+                f'{process.returncode}.\n'
+                f'Command output: {err.decode("utf-8")}')
+
+        logging.debug(f'Output:\n{out.decode("utf-8")}')
+
+    def is_file_system_node_exist(self, dir_path: str) -> bool:
+        """
+        Check for the existence of the file system node.
+        :param dir_path: file or directory path
+        :return: bool
+        """
+        return path.exists(dir_path)
+
+    def is_inside_hare(self) -> bool:
+        """
+        Check current working directory is hare home directory or not.
+        if not then raise the exception.
+        :return: None
+        """
+        if getcwd().endswith(f"{sep}hare") is False:

```

Comment:
Why do we have this check? What problem are we trying to avoid?

Reply:
https://github.com/Seagate/cortx-hare/pull/1734#discussion_r693877047

Reply:
Done